wandb/sdk/wandb_init.py

```diff
@@ -270,7 +270,9 @@ def _jupyter_teardown(self):
         ipython = self.notebook.shell
         self.notebook.save_history()
         if self.notebook.save_ipynb():
-            self.run.log_code(root=None)
+            art = self.run.log_code(root=None)
+            # explicitly clean up artifact directory here to prevent garbage collection from issuing errors
+            art._artifact_dir.cleanup()
```

Comment:
A better change would be to add a private cleanup function into the artifact object
```
art._cleanup_create_artifact_dir()
```
That way if artifact adds a better interface it will be more obvious that someone is depending on this internal datastructure.

Reply:
Good call. Fixed